{{cookiecutter.project_slug}}/apps/pages/blocks.py

```diff
@@ -1,36 +1,320 @@
+from django.core.exceptions import ValidationError
+from django.forms.utils import ErrorList
+
+from wagtail.contrib.table_block.blocks import TableBlock as WagtailTableBlock
 from wagtail.core.blocks import (
-    BooleanBlock, CharBlock, ChoiceBlock, RawHTMLBlock, RichTextBlock, StreamBlock, StructBlock,
-    URLBlock
+    BooleanBlock, CharBlock, ChoiceBlock, DecimalBlock, ListBlock, PageChooserBlock, RawHTMLBlock,
+    RichTextBlock, StreamBlock, StructBlock, StructValue, URLBlock
 )
+from wagtail.documents.blocks import DocumentChooserBlock
+from wagtail.embeds.blocks import EmbedBlock as WagtailEmbedBlock
 from wagtail.images.blocks import ImageChooserBlock
 
 
-class ImageBlock(StructBlock):
+class BaseBlock(StructBlock):
+
+    class Meta:
+        form_classname = 'full-width'
+        form_template = 'admin/blocks/base_block.html'
+
+
+class AdvancedBlock(StructBlock):
+    """
+    Advanced fields must be declared after all other fields and the order of advanced_fields
+    in class Meta must match the order of these.
+    """
+
+    class Meta:
+        form_classname = 'full-width'
+        form_template = 'admin/blocks/advanced_block.html'
+
+
+class CustomURLValue(StructValue):
+    """
+    Get active link used in CustomURL or CustomURLButton if there is one
+    """
+
+    def get_url(self):
+        link_to = self.get('link_to')
+
+        if link_to == 'page' or link_to == 'file':
+            return self.get(link_to).url
+        elif link_to == 'custom_url':
+            return self.get(link_to)
+        return None
+
+
+class CustomURLBlock(StructBlock):
+    link_to = ChoiceBlock(
+        choices=[
+            ('page', 'Page'),
+            ('file', 'File'),
+            ('custom_url', 'Custom URL'),
+        ],
+        required=False,
+    )
+    page = PageChooserBlock(required=False)
+    file = DocumentChooserBlock(required=False)
```

Comment:
Could we name this document?

It's a Python built-in, and always feels a bit weird when it gets highlighted.

I don't mind the label being called file if that's more natural though!